source/Logger.cpp

```diff
@@ -0,0 +1,42 @@
+/* Logger.cpp
+Copyright (c) 2022 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Logger.h"
+
+#include <iostream>
+#include <mutex>
+
+using namespace std;
+
+namespace {
+	std::function<void(const string &message)> logErrorCallback = nullptr;
+	mutex logErrorMutex;
+}
+
+
+
+void Logger::SetLogErrorCallback(std::function<void(const string &message)> callback)
```

Comment:
You're `using namespace std` at the top so this shouldn't be necessary? (It's not used on the `string` in this line.)
```suggestion
void Logger::SetLogErrorCallback(function<void(const string &message)> callback)
```